src/cplscheme/config/CouplingSchemeConfiguration.cpp

```diff
@@ -979,12 +979,6 @@ void CouplingSchemeConfiguration::addDataToBeExchanged(
     if (from == accessor) {
       scheme.addDataToSend(exchange.data, exchange.mesh, requiresInitialization);
     } else if (to == accessor) {
-      // Checks for a serial coupling scheme, where initial data is received by first participant.
-      if (!scheme.doesFirstStep() && requiresInitialization && (_config.type == VALUE_SERIAL_EXPLICIT || _config.type == VALUE_SERIAL_IMPLICIT) && getWaveformUsedOrder(accessor, dataName) == 0) // the order of the statements is crucial! getWaveformUsedOrder should be called last, because it may run into an error, if the data is initialized, but not defined as read data (for example Integration/Serial/AitkenAcceleration)
-      {
-        PRECICE_WARN("In serial coupling, initialized data received by the second participant will only be used if a waveform order > 0 is set. You can avoid unnecessary data initialization by setting initialize=\"false\" in the <exchange data=\"{}\" mesh=\"{}\" from=\"{}\" to=\"{}\" initialize=\"{}\" /> tag in the <coupling-scheme:... /> of your precice-config.xml. If you want to use the initial data, please consider increasing the waveform-order in the <read-data name=\"{}\" mesh=\"{}\" /> tag in the <participant ... /> named \"{}\". ",
-                     dataName, meshName, from, to, requiresInitialization, dataName, meshName, to);
-      }
```

Comment:
Why can you remove this here?

Reply:
Good question. I think this change should already have been implemented in #1422. I will try to remove it from this PR and put it into an independent PR/commit. I first have to check this branch and do some experiments to make sure. 

The important thing is the following: zeroth-order interpolation has a jump at the border of two windows. In pre-subcycling and pre-waveform times we did not really worry a lot here and just assumed a constant value for the whole window. If we have two windows: window $AB$ spans $[a,b]$ and window $BC$ spans $[b,c]$, the two windows overlap at $b$. It was possible that on $AB$ and $BC$ we get different values $f_{AB}(b) \neq f_{BC}(b)$ at $b$, because there was just a constant value stored on each window.

With #1422 this changed, because the values get a time stamp and therefore we need consistent behavior. Meaning: We have to decide on one value at b that we use on $AB$ and $BC$. Here, I decided that for zeroth order interpolation $t \in (a,b]$ gives the value associated with $b$ and $t\in (b, c]$ gives the values associated with $c$. Important question: What value do we get at $a$? If we don't initialize the data (independent from the coupling scheme) we have a gap here (set $f(x) = 0$). If we want to allow a value at $f(a) \neq 0$, we have to be able to initialize data for all participants and all coupling schemes.

Important remark: All this is only visible for a user that explicitly asks for the value at the beginning of the window by calling `read...Data` with `relativeDt = 0`. Since `relativeDt` defaults to the end of the window a user that does not use the (experimental) waveform feature will not have to worry about any of these technicalities.

Reply:
I added a PR: https://github.com/precice/precice/pull/1455 and a todo above to first conclude on https://github.com/precice/precice/pull/1455.

@uekerman feel free to resolve this conversation.